core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/PropertiesUtils.java

```diff
@@ -468,8 +468,8 @@ public static void updateURIParams(AbstractCamelModelElement selectedEP, Paramet
 			selectedEP.setParameter("uri", newUri);
 		} else {
 			// normal uri options
-			boolean valueDeleted = value == null || value.toString().trim().length() < 1
-					|| (p.getDefaultValue() != null && value.toString().equals(p.getDefaultValue()));
+			boolean equalsDefaultValue = p.getDefaultValue() != null && value.toString().equals(p.getDefaultValue());
+			boolean valueDeleted = (value == null) || value.toString().trim().length() < 1 || (equalsDefaultValue && AbstractCamelModelElement.useOptimizedXML());
```

Comment:
you are checking for a null 'value' although the line before value.toString is called